camel-core/src/main/java/org/apache/camel/model/RouteDefinition.java

```diff
@@ -1132,6 +1169,17 @@ protected RouteContext addRoutes(CamelContext camelContext, Collection<Route> ro
             }
         }
 
+        // configure Log EIP mask
+        if (logMask != null) {
+            Boolean isLogMask = CamelContextHelper.parseBoolean(camelContext, getLogMask());
+            if (isLogMask != null) {
+                routeContext.setLogMask(isLogMask);
+                if (isLogMask) {
+                    log.debug("Security mask for Logging is enabled on route: {}", getId());
```

Comment:
Should it be only context level? I added this as all other options in [RuntimeConfiguration](https://github.com/apache/camel/pull/1561/files#diff-936d95845082ad400b2c91c6b14e17c4) can be overridden at route level. It's only effective for log EIP though. Do you think we also remove the logMask flag from log component option as well so that the flag could be set only at context level?